File: 2_Regression/ARX_Regression/empirical_id.py
import numpy as np
import apm_id as arx

######################################################
# Configuration
######################################################
# number of terms
ny = 2 # output coefficients
nu = 1 # input coefficients
# number of inputs
ni = 1
# number of outputs
no = 1
# load data and parse into columns
data = np.loadtxt('data_step_test.csv',delimiter=',')
######################################################

# generate time-series model
arx.apm_id(data,ni,nu,ny)

File: 2_Regression/Higher_order_MIMO/APM_Matlab/empirical_id.m
clear all; close all; clc

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Configuration
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% number of terms
ny = 3; % output coefficients
nu = 1; % input coefficients
% number of inputs
ni = 2;
% number of outputs
no = 2;
% load data and parse into columns
load data_no_headers.csv
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% generate state space model
addpath('apm')
sysd = apm_id(data_no_headers,ni,nu,ny);